src/x/sync/types.go

```diff
@@ -88,6 +89,9 @@ type WorkerPool interface {
 
 	// GoWithTimeoutInstrument instruments GoWithTimeout with timing information.
 	GoWithTimeoutInstrument(work Work, timeout time.Duration) ScheduleResult
+
+	// GoWithCtx waits until a worker is available or the provided ctx is canceled.
+	GoWithCtx(ctx context.Context, work Work) ScheduleResult
```

Comment:
nit: Other naming usually only uses abbreviations in local vars or struct fields and the type itself is fully spelled out. Guess you could call it Java-iesque naming... but anyhow would be slightly more consistent to call this `GoWithContext(...)`.